osu.Game/Overlays/BeatmapListing/BeatmapSearchFilterRow.cs

```diff
@@ -79,9 +81,30 @@ public BeatmapSearchFilter()
 
                 TabContainer.Spacing = new Vector2(10, 0);
 
-                if (typeof(T).IsEnum)
+                var type = typeof(T);
+
+                if (type.IsEnum)
                 {
-                    foreach (var val in (T[])Enum.GetValues(typeof(T)))
+                    if (Attribute.GetCustomAttribute(type, typeof(HasOrderedElementsAttribute)) != null)
+                    {
+                        var enumValues = Enum.GetValues(type).Cast<T>().ToArray();
+                        var enumNames = Enum.GetNames(type);
+
+                        int[] enumPositions = Array.ConvertAll(enumNames, n =>
+                        {
+                            var orderAttr = (OrderAttribute)type.GetField(n).GetCustomAttributes(typeof(OrderAttribute), false)[0];
```

Comment:
Could get all fields at once to save on reflection runtime. If an `OrderAttribute` is missing the numeric enum value could be used to order.